privacy_evaluator/models/torch/_train_dcti.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+
+import torch
+import torch.nn as nn
+import torch.optim as optim
+
+import torchvision
+import torchvision.datasets as datasets
+import torchvision.transforms as transforms
+
+from torch.utils.data import DataLoader
+
+from dcti import *
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+def train(net, loader, optimizer, criterion, epoch):
+    ls = []
+    net.train()
+    for batch_idx, (data, target) in enumerate(loader):
+        optimizer.zero_grad()
+
+        data = data.to(device)
+        target = target.to(device)
+
+        output = net(data)
+        loss = criterion(output, target)
+        loss.backward()
+        optimizer.step()
+
+        ls.append(loss.item())
+
+    return torch.tensor(ls).mean()
+
+
+def test(net, loader):
+    net.eval()
+    batch_size = loader.batch_size
+    prediction = torch.zeros(len(loader.dataset))
+    with torch.no_grad():
+        for i, (data, _) in enumerate(loader):
+            data = data.to(device)
+            output = net(data).argmax(dim=1).cpu()
+            prediction[i * batch_size : i * batch_size + len(output) :] = output
+
+    return prediction
+
+
+def main():
+    transform_train = transforms.Compose(
```

Comment:
Please add a comment here why you perform these operations

Reply:
This is actually code from team 1 and they fixed it as far as we know.